### ISSUE TITLE
What has changed?

### ISSUE THREAD

**Issue by salibaj2** [FrSkyRC/ETHOS-Feedback-Community#3233]
Hi,

I build my own sensors, and as other posts and threads, after 1.4.9 update, (I think), I cannot get my sensors to work as they should.

I have same software on the MCU side but somehow, I can see the left hand dots in the telemetry screen, being updated slowly, which means that I am loosing a lot of time.

What shall I do to fix this issue please? Or else, is it intended to be fixed in the coming updates?

Again, I had a nice telemetry functioning and now I am feeling I am wasting a lot of time looking for something beyond me.

Your help in the matter is appreciated.

Thank you

**Comment by mawzthefinn** [comment]
Out of bounds checking got tightened up around then IIRC. If you are using FrSky sensor types, make sure your data is within the same value boundaries.

Also some of the newer receivers are decidedly pickier about S.port timing than older receivers. Essentially any receiver which can do FBus and S.Port at the same time (ie any one with channel ports which can be set to Fbus, S.Port and SBus as well as PWM channels), I believe this is to ensure that S.port telemetry can be properly encapsulated in FBus prior to final processing.

**Comment by salibaj2** [comment]
Thanks as always mawzthefinn.

Not sure if I understood the out of bounds checking.

Can you send me an email please? my username @gmail.com?

Thanks

**Comment by salibaj2** [comment]
bsonigs,

I am working to verify that the problem is not by my side. Will get to you later please so as to waste no time to you.

I am revisiting my code to make it more efficient and will let you know. Thanks goes to mawz for his help as well.

**Comment by bsongis-frsky** [comment]
@salibaj2 everything works correctly?

**Comment by salibaj2** [comment]
Yes, looks ok. After updated TD-ISRM to 2.2.6 and updated all receivers, I begin to having this issue.

I only noticed that after updating the transmitter, and it told me that update was successful, the truth was that ISRM did not update at all, it stayed on 2.2.4.

I noticed this by mistake, as I went to info and saw that ISRM was still on 2.2.4 although I did the update.

I tried two more times for the update to go through, as even on the second attempt, it reported as updated and it was not.

This happened while I was on v1.4.17, now moved to 1.5.7.

Although I have signed the Non-disclosure Agreement, and have all the documentation, I don't know where I can post or ask about telemetry, as FrSky never gave me a link, or that is how it should be.

For me telemetry is good, but I think that it still need some fine tuning.

For example, why if I try to rediscover sensor without deleting them, I end up with multiple sensors of the same type?

If I happen to send same sensor faster than required, why same sensor is discovered twice while discovering sensors?

It is ok for a computer literate person but who is not able to understand these things people keep asking the same questions.

Thank you for your attention and help.
